pkg/resource/hyper_parameter_tuning_job/custom_delta.go

```diff
@@ -32,4 +32,35 @@ func customSetDefaults(
 			}
 		}
 	}
+
+	// TODO: Use late initialize instead once code generator supports late initializing slices.
+	if ackcompare.IsNotNil(a.ko.Spec.TrainingJobDefinitions) && ackcompare.IsNotNil(b.ko.Spec.TrainingJobDefinitions) {
+		if len(a.ko.Spec.TrainingJobDefinitions) == len(b.ko.Spec.TrainingJobDefinitions) {
+			for i := 0; i < len(a.ko.Spec.TrainingJobDefinitions); i++ {
```

Comment:
use for loop with range clause, its like for each loop

sample: https://github.com/aws-controllers-k8s/sagemaker-controller/blob/main/pkg/resource/training_job/custom_delta.go#L31

Reply:
done